test/defer.js

```diff
@@ -0,0 +1,211 @@
+          data: null,
+          errors: [{
+            message: 'Server received an operation that uses incremental delivery (@defer or @stream), but the client does not accept multipart/mixed HTTP responses. To enable incremental delivery support, add the HTTP header "Accept: multipart/mixed; deferSpec=20220824".'
+          }]
+        })
+      })
+
+      await app.close()
+      t.end()
+    })
+  }
+
+  const correctAcceptValues = [
+    'multipart/mixed; deferSpec=20220824',
+    'multipart/mixed; deferSpec=20220824, application/json',
+    'application/json, multipart/mixed; deferSpec=20220824'
+  ]
+
+  for (const accept of correctAcceptValues) {
+    test('works with @defer when used with correct "accept" header', async t => {
+      const app = Fastify()
+      await app.register(mercurius, { schema, resolvers, graphiql: true, defer: true })
+
+      const res = await app.inject({
+        method: 'POST',
+        url: '/graphql',
+        headers: {
+          'content-type': 'application/json',
+          accept
+        },
+        body: JSON.stringify({ query })
+      })
+
+      t.match(res, {
+        statusCode: 200,
+        headers: {
+          'content-type': 'multipart/mixed; boundary="-"; deferSpec=20220824'
+        },
+        body: `\r
+---\r
+content-type: application/json; charset=utf-8\r
+\r
+{"hasNext":true,"data":{"allProducts":[{"delivery":{},"sku":"sku","id":"0"}]}}\r
+---\r
+content-type: application/json; charset=utf-8\r
+\r
+{"hasNext":false,"incremental":[{"path":["allProducts",0,"delivery"],"data":{"estimatedDelivery":"25.01.2000","fastestDelivery":"25.01.2000"}}]}\r
+-----\r
+`
+      })
+
+      await app.close()
+      t.end()
+    })
+  }
+
+  test('returns stream when using undici.fetch with @defer', async t => {
+    const app = Fastify()
+    await app.register(mercurius, { schema, resolvers, graphiql: true, defer: true })
+    const url = await app.listen({ port: 0 })
+
+    const res = await fetch(`${url}/graphql`, {
+      method: 'POST',
+      headers: {
+        'content-type': 'application/json',
+        accept: 'multipart/mixed; deferSpec=20220824'
+      },
+      body: JSON.stringify({ query })
+    })
+
+    const reader = res.body.getReader()
+    const { value } = await reader.read()
+    const result = new TextDecoder('utf-8').decode(value)
+
+    t.same(result, `\r
+---\r
+content-type: application/json; charset=utf-8\r
+\r
+{"hasNext":true,"data":{"allProducts":[{"delivery":{},"sku":"sku","id":"0"}]}}\r
+---\r
+content-type: application/json; charset=utf-8\r
+\r
+{"hasNext":false,"incremental":[{"path":["allProducts",0,"delivery"],"data":{"estimatedDelivery":"25.01.2000","fastestDelivery":"25.01.2000"}}]}\r
+-----\r
+`)
+
+    t.same(res.status, 200)
+    t.same(res.headers.get('content-type'), 'multipart/mixed; boundary="-"; deferSpec=20220824')
+
+    t.teardown(async () => {
+      await reader.releaseLock()
+      app.close()
+      process.exit()
```

Comment:
Not sure if I wasn't iterating through the stream correctly, but the test never stops - added `process.exit()` to make it passing.
@mcollina can you please help here, maybe you see a problem in my code?